pos_solidarity_rounding/static/src/js/models.esm.js

```diff
@@ -1,17 +1,20 @@
-odoo.define("pos_solidarity_rounding.models", function (require) {
-    "use strict";
+/** @odoo-module */
 
-    var models = require("point_of_sale.models");
+// SPDX-FileCopyrightText: 2025 Coop IT Easy SC
+//
+// SPDX-License-Identifier: AGPL-3.0-or-later
 
-    models.load_fields("res.partner", ["enable_solidarity_rounding"]);
+import {Order} from "point_of_sale.models";
+import Registries from "point_of_sale.Registries";
 
-    models.Order = models.Order.extend({
-        determine_tip: function () {
+const OrderRounding = (OriginalOrder) =>
+    class extends OriginalOrder {
+        determine_tip() {
             var amount = this.get_total_with_tax();
             var target_amount = Math.ceil(amount);
             return target_amount - amount;
-        },
-        tip_exists: function () {
+        }
+        tip_exists() {
             var lines = this.get_orderlines();
```

Comment:
use `const` instead of `var`.

the `for` loop below should be rewritten as:

```js
if (lines.any((line) => line.get_product() === tip_product)) {
    return true;
}
```